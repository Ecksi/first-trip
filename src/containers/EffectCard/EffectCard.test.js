import React from 'react';
import { shallow } from 'enzyme';
import { EffectCard, mapStateToProps, mapDispatchToProps } from './index';

describe('EffectCard', () => {
  let effectCard;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      filters: ['happy', 'tired'],
      effect: 'positive',
      addFilters: jest.fn(),
      removeFilter: jest.fn(),
    };

    effectCard = shallow(<EffectCard  {...mockProps} />);
  });

  it('should match the snapshot', () => {
    expect(effectCard).toMatchSnapshot();
  });

  it('should listen for a click', () => {
    let mockEvent = {'target': { 'name': 'happy' }};
    let expected = 'happy';

    effectCard.instance().handleClick(mockEvent);

    expect(mockProps.removeFilter).toHaveBeenCalledWith(expected);
  });

  describe('mapStateToProps', () => {
    it('can access all filters from the store', () => {
      const filters = mockProps.filters;
      const mockState = { filters };
      const mappedProps = mapStateToProps(mockState);
      const expected = ['happy', 'tired'];

      expect(mappedProps.filters).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls addFilters dispatch with the correct params', () => {
      const mockDispatch = jest.fn();
      const filters = ['happy', 'tired'];
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'ADD_FILTER',
        filters
      };

      mappedProps.addFilters(filters);

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('calls removeFilters dispatch with the correct params', () => {
      const mockDispatch = jest.fn();
      const filter = 'tired';
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'REMOVE_FILTER',
        filter
      };

      mappedProps.removeFilter(filter);

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  });
});