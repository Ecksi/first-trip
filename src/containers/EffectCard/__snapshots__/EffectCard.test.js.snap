// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EffectCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    addFilters={[Function]}
    effect="positive"
    filters={
        Array [
            "happy",
            "tired",
          ]
    }
    removeFilter={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        name="positive"
        onClick={[Function]}
>
        positive
</button>,
      "className": "effect-button null",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "positive",
        "name": "positive",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "positive",
      "type": "button",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          name="positive"
          onClick={[Function]}
>
          positive
</button>,
        "className": "effect-button null",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "positive",
          "name": "positive",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "positive",
        "type": "button",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
